examples/inspection/plot_model_inspection_cross_validation.py

```diff
@@ -0,0 +1,235 @@
+# during cross-validation.
+#
+# We observe a little variation in terms of R2 score. Thus, we can now safely
+# interpret if the results obtained are synonymous of having a good model.
+#
+# Our baseline perform around an R2 of 0.6 which is better than a dummy
+# regressor. Therefore, such a predictive model can be used as a baseline if
+# we would like to develop more advanced machine learning pipelines.
+#
+# To conclude, cross-validation allows to answer to two questions here: are the
+# results reliable and, if it is the case, how good is my predictive model.
+#
+# Model inspection
+# ................
+#
+# Once we are happy or at least aware of our model limitation, we can
+# investigate the internals of our model. When we performed the
+# cross-validation, we pass a parameter `return_estimator` to `True`.
+#
+# It allows to get the different predictive models trained and tested within
+# cross-validation.
+
+# %%
+cv_estimators = cv_results["estimator"]
+
+# %%
+# While the cross-validation allows us to know if we got reliable models, we
+# did not check if the model at each round were similar.
+#
+# We recall that in our case, our baseline was required to tune tha parameter
+# `alpha`. Therefore, up-to-now, we do not know if the optimal `alpha`
+# parameter for all models are similar. We can reformulate this as: what is the
+# variance of the `alpha` parameter values across iterations.
+#
+# Indeed, if the parameters `alpha` are varying depending of the input data, it
+# might be challenging to put our model in production because we will not be
+# able to fix this hyperparameter.
+#
+# Let's check the `alpha` parameter variance.
+
+# %%
+alpha = [est[-1].alpha_ for est in cv_estimators]
+plt.hist(alpha, bins=30, density=True)
+plt.xlabel("Alpha")
+plt.ylabel("Density")
+_ = plt.title("Distribution of alpha parameter \nduring cross-validation")
+
+# %%
+# We see that the regularization parameter `alpha` values are centered and
+# condensed around 40: it is a good sign. It means that most of the models
+# tuned within the cross-validation had an equivalent parameter `alpha`.
+#
+# However, not only hyperparameter such as `alpha` should be studied. The model
+# parameter coming out of the fitting process should analyzed. In our case, we
+# used a linear model. These models are parametrized and rely on two
+# parameters: `coef_` and `intercept_`. Therefore, we should analyze the
+# variance of these parameters as well.
+#
+# For the sake of simplicity, we are going to solely look at the `coef_`.
+# %%
+import pandas as pd
+
+coefficients = pd.DataFrame(
+    [est[-1].coef_ for est in cv_estimators],
+    columns=X.columns,
+)
+coefficients
+
+# %%
+import seaborn as sns
+
+plt.figure(figsize=(9, 7))
+sns.swarmplot(data=coefficients, orient="h", color="k", alpha=0.5)
+sns.boxplot(data=coefficients, orient="h")
+plt.axvline(x=0, color=".5")
+plt.title("Coefficient variability")
```

Comment:
I wonder if "Coefficient values" would be better, as the plotted data is the coefficients. Coefficient variability may make people think we are plotting some measure of variability